shell/platform/linux/fl_keyboard_manager.cc

```diff
@@ -296,20 +257,46 @@ static void responder_handle_event_callback(bool handled,
                          event_type == GDK_KEY_RELEASE);
         gdk_event_put(fl_key_event_get_origin(event));
       }
-    } else {
-      g_object_unref(pending);
     }
   }
 }
 
 static void responder_handle_embedder_event_callback(bool handled,
-                                                     gpointer user_data_ptr) {
-  responder_handle_event_callback(handled, user_data_ptr, TRUE);
+                                                     gpointer user_data) {
+  g_autoptr(FlKeyboardManagerData) data = FL_KEYBOARD_MANAGER_DATA(user_data);
```

Comment:
Should this be preceded by:

```c++
g_return_if_fail(FL_IS_KEYBOARD_MANAGER_DATA(user_data));
```

Reply:
No, those checks are only added to public methods where the caller might make a mistake. In a callback, it's safely assumed that the correct object will be provided. This will be checked by unit tests.